source/common/router/config_impl.cc

```diff
@@ -450,7 +450,7 @@ bool VirtualHostImpl::usesRuntime() const {
 VirtualHostImpl::VirtualClusterEntry::VirtualClusterEntry(
     const envoy::api::v2::VirtualCluster& virtual_cluster) {
   if (virtual_cluster.method() != envoy::api::v2::RequestMethod::METHOD_UNSPECIFIED) {
-    method_ = envoy::api::v2::RequestMethod_Name(virtual_cluster.method());
+    method_.value(envoy::api::v2::RequestMethod_Name(virtual_cluster.method()));
```

Comment:
I believe it's the [implicit copy assignment operator](http://en.cppreference.com/w/cpp/language/copy_assignment#Implicitly-declared_copy_assignment_operator).

This got caught during Google import only because of the protobuf string type mismatch, which didn't match the type defined by `Optional<std::string>`.